src/lib/run-studies.ts

```diff
@@ -1,111 +1,64 @@
 /* eslint-disable no-console */
-import {
-    ECSClient,
-    LaunchType,
-    RunTaskCommand,
-    RunTaskCommandInput,
-    RegisterTaskDefinitionCommand,
-    RegisterTaskDefinitionCommandInput,
-    DescribeTaskDefinitionCommand,
-    DescribeTaskDefinitionCommandOutput,
-    RunTaskCommandOutput,
-} from '@aws-sdk/client-ecs'
-import { GetResourcesCommand, ResourceGroupsTaggingAPIClient } from '@aws-sdk/client-resource-groups-tagging-api'
+import { ECSClient, RunTaskCommandOutput } from '@aws-sdk/client-ecs'
+import { ResourceGroupsTaggingAPIClient } from '@aws-sdk/client-resource-groups-tagging-api'
+import { getECSTaskDefinition, registerECSTaskDefinition, runECSFargateTask, getTaskResourcesByRunId } from './aws'
 import { managementAppGetRunnableStudiesRequest, toaGetRunsRequest, filterManagmentAppRuns } from './utils'
 import 'dotenv/config'
 
 async function launchStudy(
     client: ECSClient,
     cluster: string,
     baseTaskDefinition: string,
-    subnets: string,
+    subnets: string[],
     securityGroup: string,
     runId: string,
     imageLocation: string,
     studyTitle: string,
```

Comment:
How does the SetupApp get the study title?

Reply:
Ah, that is coming [from the BMA](https://github.com/safeinsights/setup-app/blob/0c694cae50708edb225efa2ccb06c9ad198a2ce2/src/lib/types.ts#L1-L5).

Reply:
Yep -- there's no reliance on uniqueness of the value, it's purely for quality-of-life purposes in case a human / dev is looking at it.